generator/dart/generated/google_cloud_protobuf/test/timestamp_test.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:google_cloud_protobuf/protobuf.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('test zero seconds, zero nanos', () {
+    const encoding = '1970-01-01T00:00:00Z';
+    // test Timestamp.encode()
+    final timestamp = Timestamp(seconds: 0, nanos: 0);
+    expect(timestamp.toJson(), encoding);
+
+    // test Timestamp.decode()
+    final copy = Timestamp.fromJson(encoding);
+    expect(copy.seconds, 0);
+    expect(copy.nanos, 0);
+  });
+
+  test('min seconds', () {
+    final timestamp = TimestampHelper.decode('0001-01-01T00:00:00Z');
+    expect(timestamp.seconds, -62135596800);
+    expect(timestamp.nanos, 0);
+  });
+
+  test('max seconds', () {
+    final timestamp = TimestampHelper.decode('9999-12-31T23:59:59Z');
+    expect(timestamp.seconds, 253402300799);
+    expect(timestamp.nanos, 0);
+  });
+
+  // Verify timestamps can roundtrip from String -> Timestamp -> String.
+  void roundTrip(String rfc3339) {
+    test('round trip $rfc3339', () {
+      final timestamp = Timestamp.fromJson(rfc3339);
+      final encoded = timestamp.toJson();
+      expect(encoded, rfc3339);
+    });
+  }
+
+  roundTrip('0001-01-01T00:00:00Z');
```

Comment:
It would be nice to have some time with negative seconds and non-zero nanos.

Reply:
and it turns out Rust had some bugs, see #1634

Reply:
Thanks - I added the tests from the updated Rust impl and addressed an issue w/ negative seconds and non-zero nanos.